File: 99-use-cases/11-create-eval-flag/highlight-eval.js
/** Evaluate configure-grid-selection-green-highlight for grid highlight color. */

const LaunchDarkly = require("@launchdarkly/node-server-sdk");

// LaunchDarkly capability: String flag evaluation (server-side SDK)
// See: https://launchdarkly.com/docs/sdk/features/evaluations

// LaunchDarkly: flag key=configure-grid-selection-green-highlight name="Configure: grid selection green highlight" kind=boolean
// https://app.launchdarkly.com/projects/lunch-marcoly/features/configure-grid-selection-green-highlight

const FLAG_HIGHLIGHT = "configure-grid-selection-green-highlight";
const VALID_COLORS = new Set(["yellow", "red", "blue", "green", "purple"]);

function buildContext(username) {
  return { kind: "user", key: username };
}

function normalizeHighlightColor(raw) {
  const color = String(raw || "none")
    .trim()
    .toLowerCase();
  return VALID_COLORS.has(color) ? color : "none";
}

function colorLabel(highlightColor) {
  return highlightColor === "none" ? "(no-color)" : `(${highlightColor})`;
}

function buildResponse(username, raw) {
  const color = normalizeHighlightColor(raw);
  return {
    username,
    flagValue: String(raw ?? "none"),
    highlightColor: color,
    colorLabel: colorLabel(color),
  };
}

async function evaluateHighlight(client, username) {
  if (!client) {
    return buildResponse(username, "none");
  }
  const context = buildContext(username);
  const raw = await client.variation(FLAG_HIGHLIGHT, context, "none");
  return buildResponse(username, raw);
}

module.exports = {
  FLAG_HIGHLIGHT,
  buildContext,
  normalizeHighlightColor,
  colorLabel,
  buildResponse,
  evaluateHighlight,
};

File: 99-use-cases/14-progressive-rollout/dotnet/HighlightEval.cs
using LaunchDarkly.Sdk;
using LaunchDarkly.Sdk.Server;

/// <summary>
/// Evaluate the progressive-rollout highlight flag for one user context.
/// LaunchDarkly: server-side string variation.
/// https://launchdarkly.com/docs/sdk/features/evaluations
/// </summary>
sealed class HighlightEval : IDisposable
{
    internal const string FlagHighlight = "enable-grid-selection-highlight";
    private static readonly HashSet<string> ValidColors =
        new(StringComparer.OrdinalIgnoreCase) { "yellow", "red", "blue", "green", "purple" };
    private readonly LdClient? _client;

    internal HighlightEval()
    {
        var sdkKey = Environment.GetEnvironmentVariable("LD_SDK_KEY")?.Trim();
        if (string.IsNullOrEmpty(sdkKey))
        {
            Console.Error.WriteLine("Warning: LD_SDK_KEY is unset — evaluation stays at code fallback none.");
            return;
        }

        _client = new LdClient(Configuration.Builder(sdkKey).StartWaitTime(TimeSpan.FromSeconds(10)).Build());
        if (!_client.Initialized)
        {
            Console.Error.WriteLine("Warning: LaunchDarkly initialization failed — evaluation stays at code fallback none.");
            _client.Dispose();
            _client = null;
        }
    }

    internal object Evaluate(string username)
    {
        var raw = "none";
        if (_client is { Initialized: true } && !string.IsNullOrWhiteSpace(username))
            raw = _client.StringVariation(FlagHighlight, Context.New(username), "none");
        var color = ValidColors.Contains(raw.Trim()) ? raw.Trim().ToLowerInvariant() : "none";
        return new
        {
            username,
            flagValue = raw,
            highlightColor = color,
            colorLabel = color == "none" ? "(no-color)" : $"({color})",
        };
    }

    public void Dispose() => _client?.Dispose();
}
